### PR TITLE
Add 1D field plotting.

### PR DESCRIPTION
**Pull request by jb-mackay**
There may be a cleaner way to do this, but this uses `ClimaCore` tools I'm familiar with.

### CODE REVIEW

**Comment by jakebolewski** [comment]
LGTM, maybe it would be good to add a default tittle and axis labels  (ex. with the quadrature as we do for the other spectral spaces).

Test plot:
image

**Review by jakebolewski — APPROVED** [review]

**Comment by jb-mackay** [comment]
> LGTM, maybe it would be good to add a default tittle (ex. with the quadrature as we do for the other spectral spaces).

Good idea; going to add the quadrature as a label rather than a title so that we can do things like `plot(field1); plot!(field2)` and have it be interpretable.

**Review by valeriabarra — APPROVED** [review]

**Comment by valeriabarra** [comment]
Nice job, Ben!

**Comment by simonbyrne** [comment]
It might be helpful if we make the curve discontinuous at the element boundaries: it looks like the easiest  way is to insert NaNs at the discontinuities:
https://discourse.julialang.org/t/how-to-plot-correctly-a-discontinuous-function/26931/2

**Comment by jb-mackay** [comment]
@simonbyrne is this what you are envisioning? Do you just want to see the different elements distinctly? Perhaps it is possible to color them differently, but that could get confusing in the legend... 

Screen Shot 2022-01-19 at 11 03 52 PM

**Comment by valeriabarra** [comment]
If the test case is plotting a CG field, I'd actually prefer without discontinuities

**Comment by valeriabarra** [comment]
I see now no DSS

**Comment by simonbyrne** [comment]
> @simonbyrne is this what you are envisioning? Do you just want to see the different elements distinctly? Perhaps it is possible to color them differently, but that could get confusing in the legend...

There should only be a vertical gap, not a horizontal one (since we have colocated nodes at the element boundaries), so it should look continuous if the values are the same, or discontinuous if the values are different.

File: lib/ClimaCorePlots/src/ClimaCorePlots.jl [hunk 1]
```diff
@@ -13,6 +13,55 @@ import ClimaCore:
...
+        weight_idx = (Nq * (i - 1) + 1):(Nq * i)
+        vertices = Topologies.vertex_coordinates(topology, i)
+        xmin, xmax = Geometry.component(vertices[1], 1),
+        Geometry.component(vertices[2], 1)
+        coords = SVector{interpolate}(range(xmin, xmax, interpolate))
+        xcoords[interp_idx] .= coords
+        ξ, _ = Spaces.Quadratures.quadrature_points(FT, QS)
+        x = FT(0.5) * (xmin + xmax) .+ FT(0.5) * (xmax - xmin) * ξ
+        xdata[interp_idx] .=
+            Spaces.Quadratures.interpolation_matrix(coords, x) * w[weight_idx]
```
Comment (simonbyrne) [inline lib/ClimaCorePlots/src/ClimaCorePlots.jl:51]: If you want to get equispaced points, an easier option is probably the whole field to `ClosedUniform` quadrature rule, then just pull out the coordinates and values

**Comment by jakebolewski** [comment]
You can see how to do that here: https://docs.juliaplots.org/stable/generated/gr/#gr-ref39

It could be toggled with setting `interpolate=0` (which is do in other placest)

**Comment by jb-mackay** [comment]
> You can see how to do that here: https://docs.juliaplots.org/stable/generated/gr/#gr-ref39
> 
> It could be toggled with setting `interpolate=0` (which is do in other places)

Ah I see that I needed to add a NaN in the coordinate list as well

File: lib/ClimaCorePlots/src/ClimaCorePlots.jl (hunk 1 above)
Comment (jakebolewski) [reply lib/ClimaCorePlots/src/ClimaCorePlots.jl:51]: see the slice model plot

**Comment by jb-mackay** [comment]
Added `NaN`s at element boundaries; here's a FV example:
Screen Shot 2022-01-20 at 10 07 43 PM

**Comment by simonbyrne** [comment]
That looks great!

**Pull request merged**